qutebrowser/mainwindow/tabwidget.py

```diff
@@ -155,6 +180,12 @@ def update_tab_titles(self):
         for idx in range(self.count()):
             self.update_tab_title(idx)
 
+    @config.change_filter('tabs', 'title-format-pinned')
+    def update_tab_titles_pinned(self):
+        """Update all texts."""
+        for idx in range(self.count()):
+            self.update_tab_title(idx)
+
```

Comment:
Instead of defining two functions, remove the `@config.change_filter` decoration for `update_tab_titles`, make it take a `section` and `option` argument, and do something like:

```python
        if section == 'tabs' and option in ['title-format',
                                            'title-format-pinned']:
            for idx in range(self.count()):
                 self.update_tab_title(idx)
```